proto/reaction.proto

```diff
@@ -541,62 +642,84 @@ message ReactionObservation {
-  }
-  WorkupType type = 1;
+  // TODO Consider removing DISSOLUTION - collapsed into addition like
+  // CRYSTALLIZATION, TRITURATION with intention in details
+  // Consider moving repeated com
+  message WorkupStep {
+    enum WorkupStepType {
+      UNSPECIFIED = 0;
+      CUSTOM = 1;
+      // Addition (quench, dilution, extraction solvent, internal standard, etc.)
+      // Specify composition/amount in "components".
+      // Often preceded by TEMPERATURE.
+      ADDITION = 2;
+      // Change of temperature.
+      // Specify conditions in "temperature".
+      TEMPERATURE = 3;
+      // Concentration step, often using a rotovap. 
+      // Specify degree in "details" if appropriate, e.g., "to half volume".
+      CONCENTRATION = 4;
+      // Liquid extractions are often preceded by Additions. If there
+      // are multiple distinct additions prior to an extraction, it is
+      // assumed that the kept phases are pooled.
+      // Specify which phase to keep in "keep_phase".
+      EXTRACTION = 5;
+      // Filtration (can keep solid or filtrate).
+      // Specify which phase to keep in "keep phase".
+      FILTRATION = 6;
+      // Washing a solid or liquid, keeping the original phase.
+      // Specify "components" of rinse. Rinses performed in
+      // multiple stages should be given multiple workup steps
+      WASH = 7;
+      // Dried under vacuum.
+      // Specify temperature in "temperature".
+      DRY_IN_VACUUM = 8;
+      // Dried with chemical additive.
+      // Specify chemical additive in "components".
+      DRY_WITH_MATERIAL = 9;
+      // Purification by flash chromatography.
+      // Specify stationary and mobile phases (and gradients) in "details".
+      // Specify automation in "is_automated".
+      FLASH_CHROMATOGRAPHY = 10;
+      // Purification by other prep chromatography (e.g. prep TLC, prep HPLC).
+      // Specify method, stationary and mobile phases in "details".
+      // Specify automation in "is_automated".
+      OTHER_CHROMATOGRAPHY = 11;
+      // Scavenging step (e.g., pass through silica/alumina pad)
+      // Specify any material additives in "components".
+      SCAVENGING = 12;
+      // Waiting step. Specify "duration".
+      WAIT = 13;
+      // Mixing step. Specify "stirring"
+      STIRRING = 14;
+      // pH adjustments should specify "components" to define
+      // species used as well as "ph" for target ph
+      PH_ADJUST = 15;
+      // Redissolution considered to be a special form of addition.
+      // Specify "components"
+      DISSOLUTION = 16;
+      // Specify temperature in "temperature".
+      // Specify pressure and apparatus notes in "details".
+      DISTILLATION = 17;
+    }
+    WorkupStepType type = 1;
+    // Include step-specific details such as purpose of an addition, e.g.,
+    // "crystallization". Include method details, e.g., "layer diffusion".
+    string step_details = 2;
+    Time duration = 3;
+    repeated Compound components = 4;
```

Comment:
We also discussed changing `repeated Compound components = 4;` to a `ReactionInput input = 4;` so that we can include information about addition speed and addition temperature.

Reply:
When we make that change, we should also update the comments to refer to `input` not `components`

Reply:
Right, thanks. I like it.